plugin/src/withIAP.ts

```diff
@@ -0,0 +1,136 @@
+import {ConfigPlugin, createRunOncePlugin} from '@expo/config-plugins';
+import {
+  WarningAggregator,
+  withAppBuildGradle,
+  withProjectBuildGradle,
+} from '@expo/config-plugins';
+
+const pkg = require('../../package.json');
+
+type PaymentProvider = 'Amazon AppStore' | 'both' | 'Play Store';
```

Comment:
Do you think this naming is okay? I didn't put much effort into it.

Reply:
Yes, that should be OK, Thank you for your contribution